content/docs/fundamentals/converting/resolving-errors@zh_CN.md

```diff
@@ -0,0 +1,350 @@
+
+<pre class="error-text">
+必需的标记“meta name=viewport”缺失或不正确。
+</pre>
+
+AMP 要求您为视口定义 `width` 和 `minimum-scale`。这两个值必须分别定义为 `device-width` 和 `1`。视口是 HTML 网页的 `<head>` 中包含的常用标记。
+
+要解决视口错误，请将以下 HTML 代码段添加到 `<head>` 标记中：
+
+```html
+<meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+```
+
+在 AMP 中，必需为 `width` 和 `minimum-scale` 指定值。定义 `initial-scale` 不是一项强制性要求，但移动网站开发者通常都会指定此值，所以我们建议您也这么做。要详细了解视口和自适应设计，请参阅[配置视口](https://developers.google.com/speed/docs/insights/ConfigureViewport)。
+
+和之前一样，**重新加载**网页并检查错误是否已消失。
+
+## 替换外部样式表
+
+以下错误与使用样式表有关：
+
+<pre class="error-text">
+标记“link rel=stylesheet for fonts”中的属性“href”设为了无效值“base.css”。
+</pre>
+
+具体来说，该错误涉及的是 `<head>` 标记中的下列样式表链接标记：
+
+```html
+<link href="base.css" rel="stylesheet" />
+```
+
+问题在于，这是一个外部样式表引用。在 AMP 网页中，为尽可能缩短文档的加载用时，您不能添加外部样式表。相反，您必须使用 `<style amp-custom></style>` 标记在 AMP 文档中以内嵌的方式添加所有样式表规则。
+
+```html
+<style amp-custom>
+
+/* The content from base.css */
+
+</style>
+```
+
+接下来，我们来解决这个错误：
+
+1.  **移除** `<head>` 中指向外部样式表的 `<link>` 标记，然后将其替换为内嵌的 `<style amp-custom></style>` 标记。样式标记中的 `amp-custom` 属性必需提供。
+2. 将 [`base.css`](https://github.com/googlecodelabs/accelerated-mobile-pages-foundations/blob/master/base.css) 文件中的所有样式**复制**到 `<style amp-custom></style>` 标记中。
+
+和之前一样，**重新加载**网页并验证样式表错误是否已消失。
+
+{% call callout('注意', type='note') %}
+除了必须采用内嵌样式以外，所有样式信息的文件大小还不能超过 50 千字节的上限。您应该先使用 [SASS](http://sass-lang.com/) 等 CSS 预处理器压缩您的 CSS，然后再将 CSS 内嵌到 AMP 网页中。
+{% endcall %}
+
+{% call callout('重要提示', type='caution') %}
+整个 AMP 文档中只能有 1 个样式标记。如果 AMP 网页引用了多个外部样式表，则您需要将这些样式表整理为一组规则。要了解哪些 CSS 规则在 AMP 中有效，请参阅[受支持的 CSS](/zh_cn/docs/design/responsive/style_pages.html)。
+{% endcall %}
+
+## 排除第三方 JavaScript
+
+虽然对于 AMP 网页，您可以通过内嵌 CSS 的方式相对轻松地重新制作样式表，但 JavaScript 却并非如此。
+
+<pre class="error-text">
+不允许使用标记“script”（特定形式除外）。
+</pre>
+
+一般来说，只有符合以下两项主要要求的脚本才允许在 AMP 中使用：
+
+1.  所有 JavaScript 都必须是异步的（即脚本标记中包含 `async` 属性）。
+2.  JavaScript 用于 AMP 库以及网页中的任何 AMP 组件。
+
+这样实际上排除了在 AMP 中使用任何由用户生成的 JavaScript/第三方 JavaScript 的行为，如下所述的情况除外。
+
+{% call callout('注意', type='note') %}
+关于由用户生成的脚本/第三方脚本的限制只存在以下例外情况：
+
+1.  将元数据添加到网页或配置 AMP 组件的脚本。这些脚本将具有类型属性 `application/ld+json` 或 `application/json`。
+2.  iframe 中包含的脚本。在 iframe 中包含 JavaScript 应该是在万不得已时才使用的措施。应尽可能地将 JavaScript 功能替换为使用 [AMP 组件](/zh_cn/docs/reference/components.html)。在下一部分中，我们将探索我们的第一个 AMP 组件。
+{% endcall %}
+
+尝试打开外部 [`base.js`](https://github.com/googlecodelabs/accelerated-mobile-pages-foundations/blob/master/base.js) 文件。发现了什么？此文件应该不含任何 JavaScript 代码，并且只包含如下所示的信息注释：
+
+```javascript
+/*
+
+This external JavaScript file is intentionally empty.
+
+Its purpose is merely to demonstrate the AMP validation error related to the
+use of external JavaScript files.
+
+*/
+```
+
+由于此外部 JavaScript 文件并不是网站的功能性组件，因此我们可以安全地移除整个引用。
+
+从文档中**移除**以下外部 JavaScript 引用：
+
+```html
+<script type="text/javascript" src="base.js"></script>
+```
+
+现在，**重新加载**网页并验证脚本错误是否已消失。
+
+## 包含 AMP CSS 样板
+
+以下错误涉及缺失的样板代码：
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).